app/controllers/NirmsNumberController.scala

```diff
@@ -76,84 +73,37 @@ class NirmsNumberController @Inject() (
         )
   }
 
-  def onPageLoadUpdate: Action[AnyContent] =
+  def onPageLoadUpdate(mode: Mode): Action[AnyContent] =
```

Comment:
This function now allows a url hack - you can be on the maintain profile page (withhas nirms set to 'no' and no nirms number) then you can go directly to this page and put in a nirms number and it allows you to go to the CYA page and submit it even though you haven't been to the yes/no hasnirms page to change it to yes (and seen all the writing on there)

I think you shouldn't be allowed to go to this page from 'no' unless you have been through the hasNirms page journey

Reply:
Matched on request.userAnswers.getPageValue(HasNirmsUpdatePage) (the EitherNec thing) to achieve this :) - tested also.